plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -334,4 +341,26 @@ default List<String> writeNewDotPathForNestedObject(Object json, String dotPath)
     default String convertToDotPath(String path) {
         return path.replaceAll("\\[(\\d+)\\]", "$1\\.").replaceAll("\\['(.*?)']", "$1\\.").replaceAll("^\\$", "").replaceAll("\\.$", "");
     }
+
+    default void setRequestContextFromExt(SearchRequest request, PipelineProcessingContext requestContext) {
```

Comment:
this class looks more like an abstract class rather then an interface, there are verbose default implementations for every method. Consider refactoring interface->abstract class sometime in the future

Reply:
will take down this note for future refactoring in a separate PR.